bip-path-templates.mediawiki

```diff
@@ -97,29 +97,29 @@ For Unit range, we say that range_start is equal to range_end, even though there
-# If there are more than one index range within the Ranged index template, range_start of the second and any subsequent range MUST be larger than the range_end of the preceeding range.
-# For the avoidance of ambiguity, all representations of integer values larger than 0 MUST NOT start with character "0" (no leading zeroes allowed)
+# If there is more than one index range within the Ranged index template, range_start of the second and any subsequent range MUST be larger than the range_end of the preceeding range.
+# To avoid ambiguity, all representations of integer values larger than 0 MUST NOT start with character "0" (no leading zeroes allowed).
 # If hardened marker appears within any section in the path template, all preceding sections MUST also specify hardened matching.
-# For the avoidance of ambiguity, if hardened marker appears within any section in the path template, all preceding sections MUST also use the same hardened marker (either "h" or "'")
-# For the avoidance of ambiguity, trailing slashes (for example, "1/2/") and duplicate slashes (for example, "0//1") MUST NOT appear in the template
+# To avoid ambiguity, if a hardened marker appears within any section in the path template, all preceding sections MUST also use the same hardened marker (either "h" or "'").
+# To avoid ambiguity, trailing slashes (for example, "1/2/") and duplicate slashes (for example, "0//1") MUST NOT appear in the template.
 
-It may be desireable to have fully unambiguous encoding, where for each valid path template string, there are no other valid template string that matches the exact same set of paths. This would enable to compare templates for equality by simple string equality check, without any parsing.
+It may be desireable to have fully unambiguous encoding, where for each valid path template string, there is no other valid template string that matches the exact same set of paths. This would enable someone to compare templates for equality through a simple string equality check, without any parsing.
 
-To achieve full non-ambiguoussness, in addition to the rules above, one extra rule is needed:
+In addition to the rules above, one extra rule is needed:
 
 * Within Ranged index template, subsequent range MUST NOT start with the value that is equal to the end of the previous range plus one. Thus, <nowiki>"[1,2,3-5]"</nowiki> is not allowed, and should be specified as <nowiki>"[1-5]"</nowiki> instead. This rule might make templates less convenient for frequent edits, though.
 
-Also, only one type of hardened marker need to be allowed for full non-ambiguoussness (either "h" or "'"). But instead of requiring this, implementations can replace one type of marker with another in the template strings before comparing them.
+Also, only one type of hardened marker should be allowed (either "h" or "'"). But instead of requiring this, implementations can replace one type of marker with another in the template strings before comparing them.
```

Comment:
The repetitive "for full non-ambiguoussness" is added specifically to make it certain that the (either "h" or "'") rule is not a general rule, but is about this specific extra goal. I want to make it impossible to understand wrongly. I feel that because this is a separate paragraph from the previous "extra rule" paragraph, it can be a source of confusion without the explicit reference to "full non-ambigoussness"